File: wtt_web_client/src/models/Actions/Abstract/ADbHttpAction.ts
import HttpRequestMethod from "src/models/Enums/HttpRequestMethod";
import ADbWebRequest from "./ADbWebRequest";
import HttpTlsValidationMode from "src/models/Enums/HttpTlsValidationMode";

export default interface ADbHttpAction extends ADbWebRequest
{
	method: HttpRequestMethod;
	tlsValidationMode: HttpTlsValidationMode;
}

File: wtt_main_server/Models/Database/Abstract/ADbHttpAction.cs
using Models.Enums;
using Reinforced.Typings.Attributes;
using HttpRequestMethod = Models.Enums.HttpRequestMethod;

namespace Models.Database.Abstract;

#pragma warning disable CS8618

//[TsClass(IncludeNamespace = false, Order = 400)]
public abstract class ADbHttpAction : ADbWebRequest
{
	/* Метод запроса к удаленному серверу.
	 * Основной набор: GET/POST/PUT/PATCH/DELETE
	 */
	public HttpRequestMethod Method { get; set; }

	public HttpTlsValidationMode TlsValidationMode { get; set; } = 0;
}
